src/core/domain_dataclasses.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Limited
+# See LICENSE file for licensing details.
+
+"""Domain object of the Spark History Server charm."""
+
+from dataclasses import dataclass, fields
+from typing import List
+
+
+@dataclass
+class User:
+    """Class representing the user running the Pebble workload services."""
+
+    name: str
+    group: str
+
+
+@dataclass
+class S3ConnectionInfo:
```

Comment:
I believe this is just an alternative right?

Reply:
Yes, it is the dataclass abstraction I tried. But I believe to make it possible we have to resolve not only the exposing the data but also providing a way to set data. I will remove this and investigate this at a later stage